### PR TITLE
[W5.11][T15-A4]Low Jia Jin

### PR DESCRIPTION
**Pull request by lowjiajin**
I've added a comparator based sorting function that outputs a sorted list.

### CODE REVIEW

**Comment by SoilChang** [comment]
reviewed

**Comment by lowjiajin** [comment]
I’m unsure if the comparator is self evident, or whether it needs comments. Also, does it violate SLAP?

As for the first compulsory LO, I showed it in class! Not sure if you guys want me to upload it too.

**Comment by clarissayong** [comment]
Nice work! Did not think to put the sorting function in the AddressBook class, which is actually a really good idea. Learnt a lot from your code actually, like the comparing function! Thank you :-)

File: test/input.txt [hunk 1]
```diff
@@ -148,6 +148,19 @@
...
+# test sort command
+##########################################################
+  # insert in reverse order
+  add Crazy Cook p/999 e/mwahaha@gordonramsay.com a/Not in SG
+  add Billy p/44581269 e/billy@me.com a/somewhere out there
+  add Ah Bob p/97845687 e/ahbob@gmail.com a/14 Toa Payoh Lorong 4 #11-2254
+  sort
+
+  # make sure list indices work after sorting
+  delete 1
```
Comment (Esilocke) [inline test/input.txt:161]: Alternatively, list could also ensure that your list is sorted properly, but your method is fine too.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -122,6 +122,10 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
         }
     }
 
+    public void sort() {
+        this.internalList.sort(Person.nameComparator);
```
Comment (Esilocke) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: `this` not necessary here (Check the coding standards under "Avoid unnecessary use of this with fields.")

File: src/seedu/addressbook/commands/Command.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
 /**
  * Represents an executable command.
  */
-public class Command {
+public abstract class Command {
```
Comment (Esilocke) [inline src/seedu/addressbook/commands/Command.java:14]: Do we really need to make Command abstract here?

**Review by Esilocke — COMMENTED** [review]
Documentation (user guide + help) is missing, so please add those in before the next tutorial.

File: src/seedu/addressbook/commands/Command.java (hunk 3 above)
Comment (lowjiajin) [reply src/seedu/addressbook/commands/Command.java:14]: Shouldn't we though? It helps dummy proof our code. After all, instantiating a Command object with no actual implementation of functionality is rather useless.

Comment (Esilocke) [reply src/seedu/addressbook/commands/Command.java:14]: Well yes, that's certainly true. In that case though the execute method could be removed as it doesn't serve any purpose..

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 2 above)
Comment (lowjiajin) [reply src/seedu/addressbook/data/person/UniquePersonList.java]: Fixed. Modified the user guide and added the javadocs too (see latest commit), missed that it was required.

**Pull request closed without merging**